boot/environ/CMakeLists.txt

```diff
@@ -105,8 +105,12 @@ target_link_libraries(bootmgfw bootlib cportlib cmlib rtl libcntpr)
 
 if(STACK_PROTECTOR)
     target_link_libraries(bootmgfw gcc_ssp)
-elseif(RUNTIME_CHECKS)
-    target_link_libraries(bootmgfw runtmchk)
+endif()
+
+if(RUNTIME_CHECKS)
+    if(MSVC)
```

Comment:
Having to do this check everywhere is messy?
I'd just add a function: `ROS_ADD_RUNTIME_CHECKS(target)` and in that function you check if runtime checks are enabled, if they are not ofc do nothing,
if they are enabled you check compiler, and change compiler specific crap.
This way when adding this, we dont have to edit the entire tree, but can simply update this cmake function for new checks.

Reply:
Optionally, can add this function in `sdk/cmake/gcc.cmake` etc, so that each compiler specific file can add it's own checks.

Reply:
(and no need to prefix the function with "ROS_": all of our own cmake functions/macros don't have this name as well)

Reply:
Or maybe we could add an empty runtmchk library for non-MSVC targets so that it can be used unconditionally.